### PR TITLE
bump Gateway API v1.5.0-rc.1

### PR DESCRIPTION
**Pull request by zirain**
- Move test to a separated package to decoupe the `sigs.k8s.io/gateway-api` and `sigs.k8s.io/gateway-api/conformance`
- Move `internal/gatewayapi/conformace` under `test`, it's not used, and production shouldn't rely on conformance.

### CODE REVIEW

File: internal/gatewayapi/xlistenerset.go [hunk 1]
```diff
@@ -71,7 +71,6 @@ func (t *Translator) processXListenerSet(xls *gwapixv1a1.XListenerSet, gatewayMa
 		// Initialize listener status conditions
 		xls.Status.Listeners = append(xls.Status.Listeners, gwapixv1a1.ListenerEntryStatus{
 			Name:           listener.Name,
-			Port:           listener.Port,
```
Comment (zirain) [inline internal/gatewayapi/xlistenerset.go:74]: related to https://github.com/kubernetes-sigs/gateway-api/pull/4426

File: api/v1alpha1/envoygateway_types.go [hunk 2]
```diff
@@ -134,14 +134,9 @@ type EnvoyGatewaySpec struct {
 
 // GatewayAPI defines an experimental Gateway API resource that can be enabled.
 // +enum
-// +kubebuilder:validation:Enum=XListenerSet;XBackendTrafficPolicy
```
Comment (zirain) [inline api/v1alpha1/envoygateway_types.go:137]: XListenerSet promote to `v1` in Gateway API v1.5

File: internal/cmd/egctl/testdata/translate/out/rejected-http-route.route.yaml [hunk 3]
```diff
@@ -1,90 +0,0 @@
...
-          from: Same
-      name: tls
-      port: 8443
-      protocol: TLS
-  status:
-    listeners:
-    - attachedRoutes: 0
-      conditions:
-      - lastTransitionTime: null
-        message: Listener must have TLS set when protocol is TLS.
```
Comment (zirain) [inline internal/cmd/egctl/testdata/translate/out/rejected-http-route.route.yaml:35]: Gateway API add CEL [validation](https://github.com/kubernetes-sigs/gateway-api/blob/e0bee71fc23d471908564765ba2eda1eacebd78a/apis/v1/gateway_types.go#L240) for this.

File: test/conformance/suite.go [hunk 4]
```diff
@@ -16,6 +16,28 @@ import (
...
+		tests.GatewayWithAttachedRoutes,
+		tests.GatewayTLSBackendClientCertificate,
+		tests.GatewayFrontendClientCertificateValidation,
+		tests.GatewayInvalidFrontendClientCertificateValidation,
+		tests.GatewayFrontendInvalidDefaultClientCertificateValidation,
+		tests.HTTPRouteHTTPSListenerDetectMisdirectedRequests,
+		tests.HTTPRoute303Redirect,
+		tests.HTTPRoute307Redirect,
+		tests.HTTPRoute308Redirect,
+		tests.HTTPRouteHostnameIntersection,
```
Comment (zirain) [inline test/conformance/suite.go:40]: TODO: Tracked in seperated issue.

File: charts/gateway-helm/templates/_rbac.tpl [hunk 5]
```diff
@@ -9,8 +9,6 @@ All namespaced resources for Envoy Gateway RBAC.
 - {{ include "eg.rbac.namespaced.gateway.envoyproxy.status" . | nindent 2 | trim }}
 - {{ include "eg.rbac.namespaced.gateway.networking" . | nindent 2 | trim }}
 - {{ include "eg.rbac.namespaced.gateway.networking.status" . | nindent 2 | trim }}
-- {{ include "eg.rbac.namespaced.gateway.networking.experimental" . | nindent 2 | trim }}
```
Comment (arkodg) [inline charts/gateway-helm/templates/_rbac.tpl:12]: curious why these are removed

File: internal/cmd/egctl/translate_test.go [hunk 6]
```diff
@@ -212,14 +212,6 @@ func TestTranslate(t *testing.T) {
 			resourceType: string(EndpointEnvoyConfigType),
 			expect:       true,
 		},
-		{
```
Comment (arkodg) [inline internal/cmd/egctl/translate_test.go:215]: curious why

File: api/v1alpha1/envoygateway_types.go (hunk 2 above)
Comment (arkodg) [reply api/v1alpha1/envoygateway_types.go:137]: guessing this is the breaking change because even if a user wants to stay on XListenerSet, then cant ? and must migrate to ListenerSet when upgrading to v1.8 ?

File: embed.go [hunk 7]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 var (
-	//go:embed charts/gateway-helm/crds/gatewayapi-crds.yaml
+	//go:embed gatewayapi-crds.yaml
```
Comment (arkodg) [inline embed.go]: cuirous why are we moving this out of chart

**Comment by arkodg** [comment]
hey thanks for driving this work !
this one PR is trying to do a lot, is it possible to split up into smaller parts ?

is the plan to rm all CRDs from `gateway-helm` and recommend users to use `gateway-crds-helm` starting v1.8 ?
cc @envoyproxy/gateway-maintainers

File: api/v1alpha1/envoygateway_types.go (hunk 2 above)
Comment (zirain) [reply api/v1alpha1/envoygateway_types.go:137]: yes, If we need rethink about this in the future, do we need to support X API.

File: internal/cmd/egctl/translate_test.go (hunk 6 above)
Comment (zirain) [reply internal/cmd/egctl/translate_test.go:215]: https://github.com/envoyproxy/gateway/pull/8161#discussion_r2807135605

File: embed.go (hunk 7 above)
Comment (zirain) [reply embed.go]: The gatewayapi-crds is large than 1MB, which make helm install never success.
Because helm store the release info in secret(the default driver), which have 1MB limitation.

**Comment by zirain** [comment]
> hey thanks for driving this work ! this one PR is trying to do a lot, is it possible to split up into smaller parts ?

It's hard, most of them are related to bump Gateway API v1.5. 
sigs.k8s.io/gateway-api/conformance now is a seperated package.

File: api/v1alpha1/envoygateway_types.go (hunk 2 above)
Comment (maelvls) [reply api/v1alpha1/envoygateway_types.go:137]: for what it's worth, we (cert-manager) have decided to do a clean cut upgrade to ListenerSet as opposed to keeping both resources. There was a discussion on whether to keep the X resources around in 1.5: https://kubernetes.slack.com/archives/CR0H13KGA/p1770328007249819

File: internal/gatewayapi/status/xlistenerset.go
Comment (maelvls) [inline internal/gatewayapi/status/xlistenerset.go:1]: Shouldn't this file be renamed to `listenerset.go`?

Comment (zirain) [reply internal/gatewayapi/status/xlistenerset.go:1]: yes, that is right.
let's do that in seperated PR.

**Comment by zhaohuabing** [comment]
> is the plan to rm all CRDs from `gateway-helm` and recommend users to use `gateway-crds-helm` starting v1.8 ?
> cc @envoyproxy/gateway-maintainers

I think keeping the CRDs in gateway-helm is still useful — at least for development and testing workflows.

Does https://github.com/envoyproxy/gateway/pull/8283 help?

File: api/v1alpha1/envoygateway_types.go (hunk 2 above)
Comment (zhaohuabing) [reply api/v1alpha1/envoygateway_types.go:137]: +1 to removing `XListenerSet` support from 1.8 since we've already set expectations in the `GatewayAPISettings` that `These APIs are experimental today and are subject to change or removal as they mature.` 

Maintaining both `XListenerSet` and `ListenerSet` would add quite a bit of ongoing overhead, and it might not be worth keeping both around.

File: embed.go (hunk 7 above)
Comment (zhaohuabing) [reply embed.go]: I think keeping the CRDs in gateway-helm is still useful — at least for development and testing workflows.

Does https://github.com/envoyproxy/gateway/pull/8283 help?

**Review by zhaohuabing — CHANGES_REQUESTED** [review]
Should we watch `TLSRoute` v1 instead of v1alpha3?

Continuing watching v1alpha3 would break standard channel users as it's not in standard channel anymore.

**Comment by zirain** [comment]
> Should we watch `TLSRoute` v1 instead of v1alpha3?
> 
> Continuing watching v1alpha3 would break standard channel users as it's not in standard channel anymore.

I could find v1alpha3 in [standard channel](sha256:ff681b2af4c2226e608360d2245e8a399a39413823f4e9c9f8ace8ec36e24b60), what I missed?

**Comment by zhaohuabing** [comment]
Is v1alpha3 TLSRoute still served in the standard channel? If it's still served, then we would be good, if not, then EG should watch v1 instead.

**Comment by zirain** [comment]
> Is v1alpha3 TLSRoute still served in the standard channel? If it's still served, then we would be good, if not, then EG should watch v1 instead.

it's still served.

File: charts/gateway-helm/templates/_rbac.tpl (hunk 5 above)
Comment (zirain) [reply charts/gateway-helm/templates/_rbac.tpl:12]: it's empty now, and will result invalid yaml file if we didn't remove this line.

Comment (arkodg) [reply charts/gateway-helm/templates/_rbac.tpl:12]: thanks, realized its tied to listenerset

File: internal/cmd/egctl/testdata/translate/out/rejected-http-route.route.yaml (hunk 3 above)
Comment (arkodg) [reply internal/cmd/egctl/testdata/translate/out/rejected-http-route.route.yaml:35]: should we then add another rejected route rule test here that adds a negative status

File: internal/gatewayapi/resource/load.go [hunk 8]
```diff
@@ -243,7 +243,7 @@ func loadKubernetesYAMLToResources(input []byte, addMissingResources bool, envoy
 					Name:      name,
 					Namespace: namespace,
 				},
-				Spec: typedSpec.(gwapiv1a3.TLSRouteSpec),
+				Spec: typedSpec.(gwapiv1.TLSRouteSpec),
```
Comment (arkodg) [inline internal/gatewayapi/resource/load.go:246]: is this needed

Comment (zirain) [reply internal/gatewayapi/resource/load.go:246]: `v1alpa3.TLSRoute` alias to `v1.TLSRoute` cause this.

**Review by arkodg — APPROVED** [review]

**Review by zhaohuabing — APPROVED** [review]
